File: java/src/test/java/challenges/aoc2024/day3/service/DayThreeSolver.java
package challenges.aoc2024.day3.service;

import java.util.List;

import challenges.aoc2024.day3.model.ComputerMemory;
import service.DaySolver;
import util.AdventUtils;

public class DayThreeSolver implements DaySolver {
    private final List<String> inputLines = AdventUtils.parseInput(2024, 3);
    private final ProgramService programService;

    public DayThreeSolver(ProgramService programService) {
        this.programService = programService;
    }

    @Override
    public String taskOne() {
        
        return String.valueOf(inputLines.stream()
            .map(ComputerMemory::of)
            .map(memory -> programService.calculateTotalOfMultiplicationInstructions(memory))
            .mapToInt(Integer::intValue)
            .sum());
    }

    @Override
    public String taskTwo() {
        return null;
    }
}

File: scala/src/main/scala/challenges/aoc2025/day3/service/DayThreeSolver.scala
package challenges.aoc2025.day3.service

import challenges.aoc2025.day3.model.BatteryBank
import service.DaySolver
import util.AdventUtils

class DayThreeSolver(batteryService: BatteryService) extends DaySolver {
  private val inputLines: List[String] = AdventUtils.parseInput(2025, 3)

  override def taskOne(): String = {
    
    val banks: List[BatteryBank] = inputLines
      .map(BatteryBank.of)
    
    batteryService.calculateTotalJoltage(banks, 2).toString
  }

  override def taskTwo(): String = {

    val banks: List[BatteryBank] = inputLines
      .map(BatteryBank.of)

    batteryService.calculateTotalJoltage(banks, 12).toString
  }
}
